npu.user.js

```diff
@@ -1014,10 +1098,159 @@ var npu = {
+							e.preventDefault();
+
+							if (!$.examSubrowsExpanded) {
+								$("img#imgcollapse_allsubrows").attr("src",
+									$("img#imgcollapse_allsubrows").attr("src").replace("plus", "minus")
+								);
+							} else {
+								$("img#imgcollapse_allsubrows").attr("src",
+									$("img#imgcollapse_allsubrows").attr("src").replace("minus", "plus")
+								);
+							}
+
+							$.examSubrowsExpanded = !$.examSubrowsExpanded;
+							changeAllSubrowVisibility($.examSubrowsExpanded);
+						});
+
+						npu.colourExams();
+						npu.hideSuccessfulExams();
+					}, 1000);
+				}
+			}, 250);
+		},
+
+		colourExams: function() {
+			$("#h_exams_gridExamList_bodytable tbody tr[hc=true]").each(function(idx, row) {
+				var lastMark = $("#h_exams_gridExamList_bodytable tbody tr.subrow#" + row.id.replace("tr__", "trs__")).first().find("table.subtable tbody tr:last td:nth-child(4)").html();
+
+				if (npu.isPassingGrade(lastMark)) {
+					$(row).addClass("npu_completed").attr("data-completed", "1");
+				} else {
+					$(row).addClass("npu_failed").attr("data-failed", "1");
+				}
+
+				$("#h_exams_gridExamList_bodytable tbody tr.subrow#" + row.id.replace("tr__", "trs__")).first().find("table.subtable tbody tr").each(function(idx, row) {
+					if (npu.isPassingGrade($(row).find("td:nth-child(4)").html())) {
+						$(row).addClass("npu_completed").attr("data-completed", "1");
+					} else {
+						$(row).addClass("npu_failed").attr("data-failed", "1");
+					}
+				});
+			});
+		},
+
+		hideSuccessfulExams: function() {
+			if(npu.getUserData(null, null, "filterExams") == "1") {
+				var subjectsToHide = [];
+				var hiddenRowCount = 0;
+
+				$("#h_exams_gridExamList_bodytable tbody tr[hc=true]").each(function(idx, row) {
+					var lastMark = $("#h_exams_gridExamList_bodytable tbody tr.subrow#" + row.id.replace("tr__", "trs__")).first().find("table.subtable tbody tr:last td:nth-child(4)").html();
+
+					if (npu.isPassingGrade(lastMark)) {
+						++hiddenRowCount;
+
+						var subjectName = $(row).find("td:nth-child(2)").find("span").first().html();
+						if ($.inArray(subjectName, subjectsToHide) === -1) {
+							subjectsToHide[subjectsToHide.length] = subjectName
+						}
+
+						$(row).remove();
+						$("#h_exams_gridExamList_bodytable tbody tr.subrow#" + row.id.replace("tr__", "trs__")).first().remove();
+					}
+				});
+
+				var rowCounter = $("#h_exams_gridExamList_tablebottom tbody tr td.grid_RowCount");
+				$(rowCounter).html(
+					$(rowCounter).html().replace(/\:(\d+)-(\d+)\/(\d+)/g, function(match, pFirst, pLast, pCount, offset, string) {
+						return pFirst + "-" + (pLast - hiddenRowCount) + "/" + (pCount - hiddenRowCount);
+					})
+				);
+
+				$.each(subjectsToHide, function(idx, subjectName) {
+					$("#upFilter_cmbSubjects").children("option").each(function() {
+						if (~$(this).html().indexOf(subjectName)) {

```

Comment:
This is a clever way of checking for `-1`, but I would prefer `!== -1` because that's more readable and easier to understand at first glance.